Content.Server/GameObjects/Components/Items/Storage/ServerStorageComponent.cs

```diff
@@ -288,12 +288,12 @@ public override void HandleNetworkMessage(ComponentMessage message, INetChannel
                         var entity = _entityManager.GetEntity(remove.EntityUid);
                         if (entity != null && storage.Contains(entity))
                         {
-                            Remove(entity);
 
                             var item = entity.GetComponent<ItemComponent>();
                             if (item != null && playerentity.TryGetComponent(out HandsComponent hands))
                             {
-                                if (hands.PutInHand(item))
+                                if (hands.CanPutInHand(item) && hands.PutInHand(item))
+                                    Remove(entity);
                                     return;
                             }
 
```

Comment:
It looks like `Remove(entity)` is not done in this case, which seems like a mistake because the `entity` is placed in the world.

Reply:
Ah, I misunderstood what Remove() actually did, I'll go ahead and fix that

Reply:
I think you misunderstood my comment - it looks like, when `TryGetComponent<HandsComponent>` fails, that `Remove` doesn't get called - because you deleted it above, and you didn't re-add it here. Is that deliberate?